modules/drivers/sensors/ltr553als/ltr553als.js

```diff
@@ -0,0 +1,342 @@
+            LED_PULSE_FREQ_MASK)
+      );
+    }
+
+    if (undefined != options.psLedDutyCycle) {
+      let value = io.readUint8(Register.PS_LED);
+      io.writeUint8(
+        Register.PS_LED,
+        (value & ~LED_DUTY_CYCLE_MASK) |
+          ((options.psLedDutyCycle << LED_DUTY_CYCLE_SHIFT) &
+            LED_DUTY_CYCLE_MASK)
+      );
+    }
+
+    if (undefined != options.psMeasRate) {
+      io.writeUint8(
+        Register.PS_MEAS_RATE,
+        options.psMeasRate & PS_MEAS_RATE_MASK
+      );
+    }
+
+    if (undefined != options.psLedPeakCurrent) {
+      let value = io.readUint8(Register.PS_LED);
+      io.writeUint8(
+        Register.PS_LED,
+        (value & ~LED_PEAK_CURRENT_MASK) |
+          (options.psLedPeakCurrent & LED_PEAK_CURRENT_MASK)
+      );
+    }
+
+    if (undefined != options.psNPulses) {
+      io.writeUint8(Register.PS_N_PULSES, options.psNPulses & PS_PULSES_MASK);
+    }
+
+    if (undefined != options.alsGain) {
+      let value = io.readUint8(Register.ALS_CONTR);
+      io.writeUint8(
+        Register.ALS_CONTR,
+        (value & ~ALS_GAIN_MASK) |
+          ((options.alsGain << ALS_GAIN_SHIFT) & ALS_GAIN_MASK)
+      );
+    }
+
+    if (undefined != options.alsIntegrationTime) {
+      let value = io.readUint8(Register.ALS_MEAS_RATE);
+      io.writeUint8(
+        Register.ALS_MEAS_RATE,
+        (value & ~ALS_INTEGRATION_TIME_MASK) |
+          ((options.alsIntegrationTime << ALS_INTEGRATION_TIME_SHIFT) &
+            ALS_INTEGRATION_TIME_MASK)
+      );
+    }
+
+    if (undefined != options.alsMeasureRate) {
+      let value = io.readUint8(Register.ALS_MEAS_RATE);
+      io.writeUint8(
+        Register.ALS_MEAS_RATE,
+        (value & ~ALS_MEASURE_RATE_MASK) |
+          ((options.alsMeasureRate << ALS_MEASURE_RATE_SHIFT) &
+            ALS_MEASURE_RATE_MASK)
+      );
+    }
+
+    if (undefined != options.psMeasRate) {
+      let value = io.readUint8(Register.ALS_MEAS_PATE_REG);
+      io.writeUint8(Register.ALS_MEAS_PATE_REG, value & PS_MEAS_RATE_MASK);
+    }
+
+    this.#setPsMode(PsMode.PS_ACTIVE_MODE);
+    this.#setAlsMode(AlsMode.ALS_ACTIVE_MODE);
+  }
+  close() {
+    this.#io.close();
+    this.#io = undefined;
+  }
+
+  get psValue() {
+    const buffer = new Uint8Array(2);
+    this.#io.readBuffer(Register.PS_DATA_LOW, buffer);
+    return (
+      ((buffer[1] & PS_DATA_HIGH_MASK) << 8) | (buffer[0] & PS_DATA_LOW_MASK)
+    );
+  }
+
+  get alsCh0Value() {
+    const buffer = new Uint8Array(2);
+    this.#io.readBuffer(Register.ALS_DATA_CH0_0, buffer);
+    return (buffer[1] << 8) | buffer[0];
+  }
+
+  get alsCh1Value() {
+    const buffer = new Uint8Array(2);
+    this.#io.readBuffer(Register.ALS_DATA_CH1_0, buffer);
+    return (buffer[1] << 8) | buffer[0];
+  }
+
+  sample() {
+    const ps = this.psValue;
+    const alsCh1 = this.alsCh1Value;
+    const alsCh0 = this.alsCh0Value;
+    const als = (alsCh0 + alsCh1) >> 1;
+
+    return {
+      ps,
+      als,
+    };
```

Comment:
This is a compound sensor. That means it has both [AmbientLight](https://419.ecma-international.org/#-sensor-classes-ambient-light) and [Proximity](https://419.ecma-international.org/#-14-sensor-classes-proximity) sensors. The form of the returned object should be:

```js
{
   proximity: {
      distance: 123
   },
   lightmeter: {
      illuminance: 234
   }
}
```